### PR TITLE
chore: Make App Sidebar Generally Available

### LINKED ISSUE #29232: [Task]: Remove feature flags for App Sidebar
Now that App Sidebar is active for all users in the latest version of Appsmith, we should remove all the feature flagging code for this and make it GA

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

Makes App Sidebar Generally Available

- Removes any feature flag control for App Sidebar
- Removes any older functionalities to handle non App Sidebar IDE versions like:
    - App Settings having an open or close state
    - Entity Explorer ability to be hidden
- Removes the announcement for App Sidebar

#### PR fixes following issue(s)
Fixes #29232

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

## Summary by CodeRabbit

- **Refactor**
	- Streamlined component logic by consolidating functions and removing unnecessary dependencies.
	- Simplified rendering and logic flow in various editor components for improved user experience.
- **Style**
	- Enhanced styling and layout in the app settings pane and editor sidebar components.
- **Chores**
	- Improved code maintainability by cleaning up unused imports, variables, and functions.
- **Bug Fixes**
	- Addressed UI inconsistencies by removing deprecated feature flags and adjusting component behavior.

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/index.tsx [hunk 1]
```diff
@@ -91,12 +90,8 @@ import { DEFAULT_ENV_ID } from "@appsmith/api/ApiUtils";
 import { isStorageEnvironmentCreated } from "@appsmith/utils/Environments";
 import type { CalloutKind } from "design-system";
 import type { FeatureFlags } from "@appsmith/entities/FeatureFlag";
-import { FEATURE_FLAG } from "@appsmith/entities/FeatureFlag";
 
-import {
-  selectFeatureFlagCheck,
-  selectFeatureFlags,
-} from "@appsmith/selectors/featureFlagsSelectors";
+import { selectFeatureFlags } from "@appsmith/selectors/featureFlagsSelectors";
```
[the remark this answers is not in the document]
Comment (hetunandu) [reply app/client/src/pages/Editor/DataSourceEditor/index.tsx:94]: @coderabbitai this is still being used

File: app/client/src/pages/Editor/Explorer/EntityExplorer.test.tsx [hunk 2]
```diff
@@ -178,6 +171,13 @@ describe("Entity Explorer tests", () => {
           <WidgetsEditorEntityExplorer />
         </MockPageDSL>,
       );
+      const widgetsTree: any = component.queryByText("Widgets", {
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/Explorer/EntityExplorer.test.tsx:174]: Looks like a unnecessary change.

Comment (hetunandu) [reply app/client/src/pages/Editor/Explorer/EntityExplorer.test.tsx:174]: This test is for the older Entity explorer and here it needs to expand the tree

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**